honeyscanner/main.py

```diff
@@ -77,8 +85,11 @@ def main() -> None:
     """
     Main entry point of the program.
     """
-    args: argparse.Namespace = parse_arguments()
     print(ascii_art_honeyscanner())
+    if not check_go_exists():
```

Comment:
This is not accurate, maybe go is installed in the system but is not in path, it's more accurate to change the error message to something like go is not found in the path. Also we need to consider go compilation on runtime, I am not sure if it's the best idea. Also, providing the compiled binary is also not the best idea, as it won't work in all architectures. Maybe we can provide for amd64 and arm64 and query the system on runtime and determine which one to use.